src/index.jsx

```diff
@@ -57,37 +61,24 @@ class ReactSwitch extends Component {
 
   $onDragStop(event) {
     const { $pos, $isDragging } = this.state;
-    const { checked, onChange, id } = this.props;
 
-    // Simulate clicking the handle
-    if (!$isDragging) {
-      this.setState({ $hasOutline: false });
-      onChange(!checked, event, id);
-      return;
-    }
-    if (checked) {
-      if ($pos > (this.$checkedPos + this.$uncheckedPos) / 2) {
-        this.setState({
-          $pos: this.$checkedPos,
-          $isDragging: false,
-          $hasOutline: false
-        });
-        return;
-      }
-      this.setState({ $isDragging: false, $hasOutline: false });
-      onChange(false, event, id);
-      return;
-    }
-    if ($pos < (this.$checkedPos + this.$uncheckedPos) / 2) {
+    if ($isDragging) {
+      event.preventDefault();
+      const halfwayCheckpoint = (this.$checkedPos + this.$uncheckedPos) / 2;
+
+      const checked = $pos > halfwayCheckpoint;
       this.setState({
-        $pos: this.$uncheckedPos,
-        $isDragging: false,
-        $hasOutline: false
+        $pos: checked ? this.$checkedPos : this.$uncheckedPos
       });
-      return;
+
+      if (event.target === this.inputRef) {
+        this.checkedStateFromDragging = checked;
+      }
+
+      this.props.onChange(checked, event, this.props.id);
```

Comment:
So the onChange function here can get called when the drag does not go past the center. This is not usually an issue, but if the user of the package has some unorthodox logic in their onChange callback it could cause issues. Could we maybe nest this in an if-statement like in the `onInputChange`-method to avoid this? ie `if (checked !== this.props.checked)` or something like that...

Reply:
Good idea. It wrapped in a conditional check now